crates/api_models/src/user/theme.rs

```diff
@@ -0,0 +1,125 @@
+use actix_multipart::form::{bytes::Bytes, text::Text, MultipartForm};
+use common_enums::EntityType;
+use common_utils::{id_type, types::theme::ThemeLineage};
+use serde::{Deserialize, Serialize};
+
+#[derive(Debug, Serialize)]
+pub struct GetThemeResponse {
+    pub theme_id: String,
+    pub theme_name: String,
+    pub entity_type: EntityType,
+    pub tenant_id: id_type::TenantId,
+    pub org_id: Option<id_type::OrganizationId>,
+    pub merchant_id: Option<id_type::MerchantId>,
+    pub profile_id: Option<id_type::ProfileId>,
+    pub theme_data: ThemeData,
+}
+
+#[derive(Debug, MultipartForm)]
+pub struct UploadFileAssetData {
+    #[multipart]
+    pub asset_name: Text<String>,
+    #[multipart(limit = "10MB")]
+    pub asset_data: Bytes,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct UploadFileRequest {
+    pub lineage: ThemeLineage,
+    pub asset_name: String,
+    pub asset_data: Vec<u8>,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct CreateThemeRequest {
+    pub lineage: ThemeLineage,
+    pub theme_name: String,
+    pub theme_data: ThemeData,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct UpdateThemeRequest {
+    pub lineage: ThemeLineage,
+    pub theme_data: ThemeData,
+}
+
+// All the below structs are for the theme.json file,
+// which will be used by frontend to style the dashboard.
+#[derive(Serialize, Deserialize, Debug)]
+#[serde(rename_all = "camelCase")]
+pub struct ThemeData {
```

Comment:
I think you should be able to keep all of the following types as private, except for `ThemeData`. Serialization / deserialization should work correctly even if the nested types are private, from what I remember.